The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12268350089/job/34230032071) [(plain)](https://github.com/rust-lang/rust/commit/263e6e173ae1df79c02ccd8bd8d7dc4615cf694d/checks/34230032071/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.768 Building wheels for collected packages: reuse
#13 2.769   Building wheel for reuse (pyproject.toml): started
#13 3.014   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.014   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.015   Stored in directory: /tmp/pip-ephem-wheel-cache-0r4ayx0c/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.017 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.412 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.413 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 3.940 Collecting virtualenv
#13 3.940 Collecting virtualenv
#13 3.975   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.040      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 68.2 MB/s eta 0:00:00
#13 4.078 Collecting distlib<1,>=0.3.7
#13 4.082   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.089      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 87.0 MB/s eta 0:00:00
#13 4.123 Collecting filelock<4,>=3.12.2
#13 4.126   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.159 Collecting platformdirs<5,>=3.9.1
#13 4.163   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.245 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.428 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.5s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      319424 kB
DirectMap2M:     8069120 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
Diff in /checkout/tests/run-make/musl-default-linking/rmake.rs:37:
             continue;
         }
 
-        let target_spec_json =
-            rustc()
-                .print("target-spec-json")
-                .target(target)
-                .arg("-Zunstable-options")
-                .run()
-                .stdout_utf8();
+        let target_spec_json = rustc()
+            .print("target-spec-json")
+            .target(target)
+            .arg("-Zunstable-options")
+            .run()
+            .stdout_utf8();
 
-        let target_spec: serde_json::Value = serde_json::from_str(&target_spec_json).expect("failed to parse target-spec-json");
+        let target_spec: serde_json::Value =
+            serde_json::from_str(&target_spec_json).expect("failed to parse target-spec-json");
         let default = &target_spec["crt-static-default"];
 
         // If the value is `null`, then the default to dynamically link from musl_base was not overriden.
Diff in /checkout/tests/run-make/musl-default-linking/rmake.rs:53:
         }
 
 
-        if default.as_bool().expect("wasn't a boolean") && !LEGACY_STATIC_LINKING_TARGETS.contains(&target) {
+        if default.as_bool().expect("wasn't a boolean")
+            && !LEGACY_STATIC_LINKING_TARGETS.contains(&target)
+        {
             panic!("{target} statically links musl libc when it should dynamically link it");
     }
     }
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/tests/run-make/pdb-sobjname/rmake.rs" "/checkout/src/tools/error_index_generator/main.rs" "/checkout/tests/run-make/long-linker-command-lines/foo.rs" "/checkout/tests/run-make/long-linker-command-lines/rmake.rs" "/checkout/src/tools/lld-wrapper/src/main.rs" "/checkout/tests/run-make/emit-path-unhashed/foo.rs" "/checkout/tests/run-make/emit-path-unhashed/rmake.rs" "/checkout/src/tools/linkchecker/main.rs" "/checkout/tests/run-make/type-mismatch-same-crate-name/crateB.rs" "/checkout/tests/run-make/type-mismatch-same-crate-name/crateA.rs" "/checkout/tests/run-make/type-mismatch-same-crate-name/rmake.rs" "/checkout/tests/run-make/type-mismatch-same-crate-name/crateC.rs" "/checkout/tests/run-make/cross-lang-lto-clang/main.rs" "/checkout/tests/run-make/cross-lang-lto-clang/rustlib.rs" "/checkout/tests/run-make/cross-lang-lto-clang/rmake.rs" "/checkout/tests/run-make/extern-flag-fun/foo.rs" "/checkout/tests/run-make/extern-flag-fun/bar.rs" "/checkout/tests/run-make/extern-flag-fun/gated_unstable.rs" "/checkout/tests/run-make/extern-flag-fun/rustc.rs" "/checkout/tests/run-make/extern-flag-fun/rmake.rs" "/checkout/tests/run-make/extern-flag-fun/bar-alt.rs" "/checkout/tests/run-make/musl-default-linking/rmake.rs" "/checkout/tests/run-make/libtest-thread-limit/test.rs" "/checkout/tests/run-make/libtest-thread-limit/rmake.rs" "/checkout/tests/run-make/cross-lang-lto/main.rs" "/checkout/tests/run-make/cross-lang-lto/rmake.rs" "/checkout/tests/run-make/cross-lang-lto/lib.rs" "/checkout/tests/run-make/compiler-lookup-paths-2/b.rs" "/checkout/tests/run-make/compiler-lookup-paths-2/c.rs" "/checkout/tests/run-make/compiler-lookup-paths-2/a.rs" "/checkout/tests/run-make/compiler-lookup-paths-2/rmake.rs" "/checkout/tests/run-make/output-with-hyphens/foo-bar.rs" "/checkout/tests/run-make/output-with-hyphens/rmake.rs" "/checkout/src/tools/linkchecker/tests/checks.rs" "/checkout/tests/run-make/lto-smoke/main.rs" "/checkout/tests/run-make/lto-smoke/rmake.rs" "/checkout/tests/run-make/lto-smoke/lib.rs" "/checkout/tests/run-make/inline-always-many-cgu/foo.rs" "/checkout/tests/run-make/inline-always-many-cgu/rmake.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-ordering/rmake.rs" "/checkout/src/tools/replace-version-placeholder/src/main.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-ordering/src/lib.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-ordering/examples/ex1.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-ordering/examples/ex2.rs" "/checkout/src/tools/wasm-component-ld/src/main.rs" "/checkout/tests/run-make/c-link-to-rust-dylib/foo.rs" "/checkout/tests/run-make/c-link-to-rust-dylib/rmake.rs" "/checkout/src/tools/html-checker/main.rs" "/checkout/tests/run-make/volatile-intrinsics/main.rs" "/checkout/tests/run-make/volatile-intrinsics/rmake.rs" "/checkout/tests/run-make/checksum-freshness/foo.rs" "/checkout/tests/run-make/checksum-freshness/rmake.rs" "/checkout/tests/run-make/checksum-freshness/lib.rs" "/checkout/src/tools/rls/src/main.rs" "/checkout/tests/run-make/pass-linker-flags-flavor/attribute.rs" "/checkout/tests/run-make/pass-linker-flags-flavor/empty.rs" "/checkout/tests/run-make/pass-linker-flags-flavor/rmake.rs" "/checkout/tests/run-make/alloc-no-sync/rmake.rs" "/checkout/src/tools/suggest-tests/src/main.rs" "/checkout/src/tools/suggest-tests/src/dynamic_suggestions.rs" "/checkout/src/tools/suggest-tests/src/tests.rs" "/checkout/src/tools/suggest-tests/src/static_suggestions.rs" "/checkout/src/tools/suggest-tests/src/lib.rs" "/checkout/tests/run-make/pdb-sobjname/main.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Wed Dec 11 02:22:46 UTC 2024
  network time: Wed, 11 Dec 2024 02:22:46 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/tests/run-make/musl-default-linking/rmake.rs
+++ b/tests/run-make/musl-default-linking/rmake.rs
@@ -1,0 +1,62 @@
+use run_make_support::{rustc, serde_json};
+
+// Please do NOT add more targets to this list!
+// Per https://github.com/rust-lang/compiler-team/issues/422,
+// we should be trying to move these targets to dynamically link
+// musl libc by default.
+static LEGACY_STATIC_LINKING_TARGETS: &[&'static str] = &[
+    "aarch64-unknown-linux-musl",
+    "arm-unknown-linux-musleabi",
+    "arm-unknown-linux-musleabihf",
+    "armv5te-unknown-linux-musleabi",
+    "armv7-unknown-linux-musleabi",
+    "armv7-unknown-linux-musleabihf",
+    "i586-unknown-linux-musl",
+    "i686-unknown-linux-musl",
+    "mips64-unknown-linux-musl",
+    "mips64-unknown-linux-muslabi64",
+    "mips64el-unknown-linux-muslabi64",
+    "powerpc-unknown-linux-musl",
+    "powerpc-unknown-linux-muslspe",
+    "powerpc64-unknown-linux-musl",
+    "powerpc64le-unknown-linux-musl",
+    "riscv32gc-unknown-linux-musl",
+    "s390x-unknown-linux-musl",
+    "thumbv7neon-unknown-linux-musleabihf",
+    "x86_64-unknown-linux-musl",
+];
+
+fn main() {
+    let targets = rustc().print("target-list").run().stdout_utf8();
+
+    for target in targets.lines() {
+        let abi = target.split('-').last().unwrap();
+
+        if !abi.starts_with("musl") {
+            continue;
+        }
+
+        let target_spec_json = rustc()
+            .print("target-spec-json")
+            .target(target)
+            .arg("-Zunstable-options")
+            .run()
+            .stdout_utf8();
+
+        let target_spec: serde_json::Value =
+            serde_json::from_str(&target_spec_json).expect("failed to parse target-spec-json");
+        let default = &target_spec["crt-static-default"];
+
+        // If the value is `null`, then the default to dynamically link from
+        // musl_base was not overriden.
+        if default.is_null() {
+            continue;
+        }
+
+        if default.as_bool().expect("wasn't a boolean")
+            && !LEGACY_STATIC_LINKING_TARGETS.contains(&target)
+        {
+            panic!("{target} statically links musl libc when it should dynamically link it");
+        }
+    }
+}

